packages/main/src/plugin/layout-manager.ts

```diff
@@ -0,0 +1,209 @@
+    };
+
+    this.#disposables.push(this.configurationRegistry.registerConfigurations([layoutConfiguration]));
+  }
+
+  dispose(): void {
+    this.#disposables.forEach(disposable => disposable.dispose());
+    this.#disposables = [];
+  }
+
+  // Set custom defaults for a table type (called from ContainerList, ImagesList, etc.)
+  setTableDefaults(tableKind: string, columnNames: string[]): void {
+    TABLE_DEFAULTS[tableKind] = columnNames;
+  }
+
+  // Get default configuration for a table type
+  getDefaultTableConfig(tableKind: string): SavedLayoutConfig[] {
+    const defaults = TABLE_DEFAULTS[tableKind] ?? [];
+    return defaults.map(name => ({ id: name, enabled: true }));
+  }
+
+  // Load table configuration (with fallback to defaults)
+  async loadTableConfig(tableKind: string, availableColumns: string[]): Promise<LayoutEditItem[]> {
+    try {
+      const config = this.configurationRegistry.getConfiguration('layout');
+      const savedConfig = config.get<SavedLayoutConfig[]>(`${tableKind}`, []);
+
+      if (!savedConfig || !Array.isArray(savedConfig)) {
+        return this.createDefaultLayoutItems(tableKind, availableColumns);
+      }
+
+      return this.mergeConfigWithAvailableColumns(tableKind, savedConfig, availableColumns);
+    } catch (error: unknown) {
+      console.warn(`Failed to load table config for ${tableKind}:`, error);
+      return this.createDefaultLayoutItems(tableKind, availableColumns);
+    }
+  }
+
+  // Save table configuration
+  async saveTableConfig(tableKind: string, items: LayoutEditItem[]): Promise<void> {
+    const configToSave: SavedLayoutConfig[] = items.map(item => ({ id: item.id, enabled: item.enabled }));
+
+    const config = this.configurationRegistry.getConfiguration('layout');
+    await config.update(`${tableKind}`, configToSave);
+  }
+
+  // Reset table to defaults
+  async resetTableConfig(tableKind: string, availableColumns: string[]): Promise<LayoutEditItem[]> {
+    const config = this.configurationRegistry.getConfiguration('layout');
+    await config.update(`${tableKind}`, undefined);
+    return this.createDefaultLayoutItems(tableKind, availableColumns);
+  }
+
+  // Helper: Create default layout items
+  private createDefaultLayoutItems(tableKind: string, availableColumns: string[]): LayoutEditItem[] {
+    const defaults = TABLE_DEFAULTS[tableKind] ?? availableColumns;
+    return defaults.map((colName, index) => ({
+      id: colName,
+      label: this.getColumnLabel(tableKind, colName),
+      enabled: true,
+      originalOrder: index,
+    }));
+  }
+
+  // Helper: Get proper label for column (dashboard sections have different labels)
+  private getColumnLabel(tableKind: string, columnId: string): string {
+    if (tableKind === 'dashboard') {
+      const labelMap: Record<string, string> = {
+        'release-notes': 'Release Notes',
+        'extension-banners': 'Extension Banners',
+        'learning-center': 'Learning Center',
+        providers: 'Providers',
```

Comment:
I think there is an issue. From what I see, there are duplicates of these constants on both backend and frontend. It'll cause issues to have duplicates. It should come from a single place and it should be registered by the component, not by a 3rd party one.